### ISSUE TITLE
how to implement loss with Intermediate data

### ISSUE THREAD

**Issue by blxl909** [open-mmlab/mmsegmentation#2617]
I am implementing ISNet(ISNet: Integrate Image-Level and Semantic-Level Context for Semantic Segmentation) on mmsegmentation, but I have some problems when implemented loss.
ISNet has two losses, one of them is LD(in paper is function 17),it's forward function needs two arguments : feature map and prediction.
I think the way mmsegmentation calculate loss is happened automatically (it's will get the predition and label to calculate loss according to source code),but I don't know how to implement loss with Intermediate data.（for example，I implement a class  A inherit from nn.Module，then I implement a class B inherit from BaseDecodeHead，and this decode head B use the class A，now I need to use A‘s loss multiply some weight  and add to final loss，how can I achieve that？）
Thanks for helping.
[ISNet.pdf](https://github.com/open-mmlab/mmsegmentation/files/10776403/ISNet.pdf)

**Comment by csatsurnh** [comment]
Hi, please refer to the Documentation [Multiple Losses](https://mmsegmentation.readthedocs.io/en/latest/tutorials/training_tricks.html#multiple-losses), also [Add new heads](https://mmsegmentation.readthedocs.io/en/latest/tutorials/customize_models.html#add-new-heads) for your information. And this issue #2225 may help.

**Comment by blxl909** [comment]
> Hi, please refer to the Documentation [Multiple Losses](https://mmsegmentation.readthedocs.io/en/latest/tutorials/training_tricks.html#multiple-losses), also [Add new heads](https://mmsegmentation.readthedocs.io/en/latest/tutorials/customize_models.html#add-new-heads) for your information. And this issue #2225 may help.

but my question is the input is not from backbone's stage , it's an intermediate outcome of some funtion . I think the current code framework can easily add loss if the input of decode head is from backbone，and I know how to implement that,but for intermediate outcome is not so directly. Is there has some easy way to accomplish my purpose？I get one answer is to reimplement backbone and return the output I need, but I think it's quiet complicated. looking forward for your help, thanks.

**Comment by MeowZheng** [comment]
we have support ISNet in project https://github.com/open-mmlab/mmsegmentation/tree/dev-1.x/projects/isnet
